Node.js/boilerplate.js

```diff
@@ -413,15 +430,99 @@ class MQBoilerPlate {
     return `${env.MQ_ENDPOINTS[i].HOST}(${env.MQ_ENDPOINTS[i].PORT})`;
   }
 
-  buildMQCNO() {
+  jwtCheck(MQDetails) {
+
+    if (!env.JWT_ISSUER) {
+
+      debug_info('JWT credentials not found, will not be using JWT to authenticate');
+      return false;
+
+    } else if (MQDetails.JWT_TOKEN_ENDPOINT === null || MQDetails.JWT_TOKEN_USERNAME === null || 
+        MQDetails.JWT_TOKEN_PWD === null || MQDetails.JWT_TOKEN_CLIENTID === null) {
+
+      debug_info('One or more JWT credentials missing, will not be using JWT to authenticate');
+      return false;
+    } 
+
+    debug_info('JWT credentials found, will be using JWT to authenticate');
+    return true;
+  }
+
+  async obtainToken() {
+    // Asynchronous function to handle http & https requests
+    let me = this;
+    let accessToken;
+
+    debug_info('Obtaining token from:', me.MQDetails.JWT_TOKEN_ENDPOINT);
+    
+    let formData = querystring.stringify({
+      username: me.MQDetails.JWT_TOKEN_USERNAME,
+      password: me.MQDetails.JWT_TOKEN_PWD,
+      client_id: me.MQDetails.JWT_TOKEN_CLIENTID,
+      grant_type: "password",
+    });
+
+    // Regex to extract the hostname:port out of the endpoint
+    // This is used to ensure the correct URL is used for the request
+    // and the JWT obtained has a valid 'iss' claim
+    let match = me.MQDetails.JWT_TOKEN_ENDPOINT.match(/^https?:\/\/([^/]+)/i);
+    let hostPort = match ? match[1] : null;
```

Comment:
I think a better name for this variable would be `hostAndPort` or something that indicates it is both. `hostPort` implies that it just holds the port of the host.